### PR TITLE
Remove empty context menu from the main menu

### PR DESCRIPTION
**Pull request by vsvyatski**
## Description
I have noticed that KeePassXC 2.3.1 has an empty context menu shown when one right-clicks on the main menu. It has a checkbox inside. If one selects/unselects the checkbox the application's main toolbar is shown/hidden. I believe this functionality is a leftover from the version 2.2.4 that was meant to be removed. In the 2.2.4 the context menu had the caption "Toolbar" and also there was _View_ > _Toolbar_ item in the main menu which is gone in 2.3.1. So I've just removed that context menu.

## Motivation and context
1. I believe that somebody just forgot to remove that context menu during the removal of _View_ > _Toolbar_.
2. Because the menu has no caption on it, it doesn't give a user much clue what it does.
3. At the moment of writing the problem is still not fixed in the develop branch.

## How has this been tested?
It is a small UI bug, so it has been tested only manually. Test environment details are below:
KeePassXC - Version 2.3.1-snapshot
Build Type: Snapshot
Revision: ddc4b9b

Libraries:
- Qt 5.10.1
- libgcrypt 1.8.2

Operating system: Manjaro Linux
CPU architecture: x86_64
Kernel: linux 4.14.31-1-MANJARO

Enabled extensions:
- Auto-Type

## Screenshots (if appropriate):
Here is how it looked before this PR
empty_menu
BTW, the same thing was on Windows
win_empty_menu
This empty menu is now gone.

## Types of changes
- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by droidmonkey** [comment]
There is an open issue wanting that to say "Show Toolbar" again. #1618

**Comment by droidmonkey** [comment]
Yes

**Comment by vsvyatski** [comment]
@droidmonkey [That](https://github.com/keepassxreboot/keepassxc/issues/1618#issuecomment-370270790) is done. The UI was implemented exactly the way it was proposed there.

File: src/gui/MainWindow.cpp [hunk 1]
```diff
@@ -841,6 +839,8 @@ void MainWindow::applySettingsChanges()
         m_inactivityTimer->deactivate();
     }
 
+    m_ui->toolBar->setHidden(config()->get("GUI/HideToolbar").toBool());
```
Comment (droidmonkey) [inline src/gui/MainWindow.cpp]: add a second parameter to the get function of "false" or add it to the default config entries in config.cpp

Comment (vsvyatski) [reply src/gui/MainWindow.cpp]: Done.

**Comment by TheZ3ro** [comment]
Thanks @vsvyatski for the contribution! :tada:

**Pull request merged**